### PR TITLE
Clarify that command-line switches need to be manually set in `addprocs`

### LINKED ISSUE #50673: Interaction between `addprocs` and `--heap-size-hint`
I think this may be a feature request/issue report rather than a question, so I am posting here rather than on the discourse.

I am basically wondering: do `Distributed.addprocs` and the CLI flag `--heap-size-hint` interact with eachother, and, if so, how? If I specify `--heap-size-hint` to the head Julia process, and then dynamically create Julia processes with `addprocs`, how (if?) does it get split up among each of the processes created?

My current workaround is to pass `--heap-size-hint` to `exeflags` in each `addprocs` call, and divvy up the memory into smaller chunks, but this seems a bit hacky.

Just wondering about this because I've ran into various segfaults which I've never been able to track down (https://github.com/JuliaLang/julia/issues/47957). I realized that memory usage seems to explode when using multiple processes which are allocated dynamically, but the memory can be controlled by setting, e.g., `--heap-size-hint` when dynamically creating the workers.

### PR DESCRIPTION
**Pull request by evetion**
Fixes #50673

also fixed a small typo in an unrelated doc page.

### CODE REVIEW

**Comment by nsajko** [comment]
There's three (!) unrelated changes in this PR. Why don't you separate the other two fixes into separate PRs?

**Comment by nsajko** [comment]
Regarding the *main* change, it's a duplicate of #50480

**Comment by evetion** [comment]
> There's three (!) unrelated changes in this PR. Why don't you separate the other two fixes into separate PRs?

I'd argue that there is a single main change, documenting how --heap-size-hint (and other flags) work with addproces. That requires adding --heap-size-hint to the docs itself.

The only unrelated change is a one character typo fix that I came across, and I didn't feel it would block this PR, so I included it. I'm happy to split it out to another PR if requested.

> Regarding the _main_ change, it's a duplicate of #50480

I missed this PR by @KnutAM, nice work! Note it's not the _main_ change in this PR, I'd say that's the interaction with addprocs.

File: doc/src/manual/command-line-interface.md [hunk 1]
```diff
@@ -131,6 +131,7 @@ The following is a complete list of command-line switches available when launchi
 |`--track-allocation[={none*\|user\|all}]` |Count bytes allocated by each source line (omitting setting is equivalent to "user")|
 |`--track-allocation=@<path>`           |Count bytes but only in files that fall under the given file path/directory. The `@` prefix is required to select this option. A `@` with no path will track the current directory.|
 |`--bug-report=KIND`                    |Launch a bug report session. It can be used to start a REPL, run a script, or evaluate expressions. It first tries to use BugReporting.jl installed in current environment and falls back to the latest compatible BugReporting.jl if not. For more information, see `--bug-report=help`.|
+|`--heap-size-hint=<size>`              |Forces garbage collection if memory usage is higher than that value. The memory hint might be specified in megabytes (500M) or gigabytes (1G).|
```
Comment (fingolfin) [inline doc/src/manual/command-line-interface.md:134]: In the meantime `--heap-size-hint` was documented at the end of this list, so this here ought to go.
```suggestion
```

**Comment by fingolfin** [comment]
Unfortunately this now has a conflict. @evetion would you be willing to resolve that. I think afterwards (and with my other comment taken into account) I think this is in a good shape for merging?

**Comment by evetion** [comment]
> Unfortunately this now has a conflict. @evetion would you be willing to resolve that. I think afterwards (and with my other comment taken into account) I think this is in a good shape for merging?

Fixed now, and moved part of the PR to the Distributed package.

**Review by inkydragon — COMMENTED** [review]
It looks like the current changes have nothing to do with `--heap-size-hint`, maybe a title change is needed, then LGTM.

**Comment by evetion** [comment]
> It looks like the current changes have nothing to do with `--heap-size-hint`, maybe a title change is needed, then LGTM.

Thanks for the suggestion, I've updated the title.

**Pull request merged**